# Philippine Data Summary Program

This notebook demonstrates the project's reusable Python modules for loading, filtering, transforming, and summarizing the Philippine Customs 2015 dataset.

**Filtering rules:**
- Quarter (`tq`) = `2015q4`
- Dutiable value (`dutiablevaluephp`) >= PHP 1,000,000

The raw `2015.csv` file should be placed in the project root folder.

In [1]:
# Import functions from the project modules
from src.data_loader import load_data
from src.processing import filter_and_transform_data
from src.summary import create_summary_tables

## Load the Dataset

In [ ]:
df = load_data()

print(f"Dataset shape: {df.shape}")
df.head()

Loading dataset: 2015.csv


c:\Users\JosefN\Desktop\Coding\2nd Year\DSA4153\Case Study 2\Philippine-Data-Summary-Program---GRP-6\src\data_loader.py:29: DtypeWarning: Columns (4: entry, 25: prefcode, 28: subport, 29: port) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(


Dataset loaded successfully: 2,236,612 rows
Dataset shape: (2236612, 30)


,uid,ty,tq,tm,entry,hscode,goodsdescription,p,q,m_fob,...,vatbase,vatpaid,othertax,finesandpenalties,dutiestaxes,prefcode,countryorigin_iso3,countryexport_iso3,subport,port
0,201501 00000001,2015,2015q1,2015m1,C,15119090000,RBD PALM OLEIN IN BULK,0.643583,2999325.0,1930315.60,...,90638835,10876660,NaN,NaN,10876660,AFTA,MYS,MYS,Sub-Port of Dumaguete,Port of Cebu
1,201501 00000002,2015,2015q1,2015m1,C,72104990000,PRIME HOT DIPPED GALVANIZED STEEL SHEET IN C,0.632000,532480.0,336527.38,...,15884083,1906090,NaN,NaN,1906090,ACFTA,CHN,CHN,Harbour Centre Port Terminal Inc,Port of Manila
2,201501 00000003,2015,2015q1,2015m1,C,72104990000,PRIME HOT DIPPED GALVANIZED STEEL SHEET IN C,0.627000,1779450.0,1115715.50,...,53064898,6367787,NaN,NaN,6367787,ACFTA,CHN,CHN,Harbour Centre Port Terminal Inc,Port of Manila
3,201501 00000004,2015,2015q1,2015m1,C,72139100000,HOT ROLLED WIRE ROD SWRY11 6.5MM,0.474000,1525249.0,722968.00,...,34664620,4159754,NaN,NaN,4501570,NaN,CHN,CHN,Harbour Centre Port Terminal Inc,Port of Manila
4,201501 00000005,2015,2015q1,2015m1,C,72163300000,STEEL STRUCTURE,1.202257,134305.2,161469.44,...,7555345,906641,NaN,NaN,906641,AKFTA,KOR,KOR,Harbour Centre Port Terminal Inc,Port of Manila


## Filter and Transform the Data

In [3]:
filtered_df = filter_and_transform_data(df)

print(f"Filtered dataset shape: {filtered_df.shape}")

filtered_df[
    [
        "countryorigin_iso3",
        "tq",
        "dutiablevaluephp",
        "dutiablevalue_million",
        "value_category"
    ]
].head()

Filtered dataset shape: (127272, 32)


,countryorigin_iso3,tq,dutiablevaluephp,dutiablevalue_million,value_category
1724767,SAU,2015q4,5153749391,5153.749391,High
2192920,KWT,2015q4,4535653502,4535.653502,High
2192918,KWT,2015q4,4434554358,4434.554358,High
2192922,KWT,2015q4,4292390824,4292.390824,High
2174779,RUS,2015q4,3595895925,3595.895925,High


## Create Summary Tables

In [4]:
# Generate the project's summary tables
summaries = create_summary_tables(filtered_df)

print("Summary tables created successfully.")

Summary tables created successfully.


### Grouped Summary

In [5]:
summaries["grouped"].head(10)

,countryorigin_iso3,record_count,total_dutiablevaluephp
0,AIA,1,3775622
1,ALB,1,1086985
2,AND,1,9257361
3,ANT,13,102405629
4,ARE,286,4148287230
5,ARG,173,6475751071
6,ASM,1,1617296
7,ATG,1,1309184
8,AUS,1387,8330212216
9,AUT,245,1317713468


### Grouped Summary by Country and Quarter

In [6]:
summaries["grouped_two"].head(10)

,countryorigin_iso3,tq,record_count,total_dutiablevaluephp
0,AIA,2015q4,1,3775622
1,ALB,2015q4,1,1086985
2,AND,2015q4,1,9257361
3,ANT,2015q4,13,102405629
4,ARE,2015q4,286,4148287230
5,ARG,2015q4,173,6475751071
6,ASM,2015q4,1,1617296
7,ATG,2015q4,1,1309184
8,AUS,2015q4,1387,8330212216
9,AUT,2015q4,245,1317713468


### Pivot Table

In [7]:
summaries["pivot"].head(10)

tq,countryorigin_iso3,2015q4
0,AIA,3775622
1,ALB,1086985
2,AND,9257361
3,ANT,102405629
4,ARE,4148287230
5,ARG,6475751071
6,ASM,1617296
7,ATG,1309184
8,AUS,8330212216
9,AUT,1317713468


### Top 10 Results

In [8]:
summaries["top10"]

,countryorigin_iso3,tq,dutiablevaluephp,dutiablevalue_million,value_category
1724767,SAU,2015q4,5153749391,5153.749391,High
2192920,KWT,2015q4,4535653502,4535.653502,High
2192918,KWT,2015q4,4434554358,4434.554358,High
2192922,KWT,2015q4,4292390824,4292.390824,High
2174779,RUS,2015q4,3595895925,3595.895925,High
1702972,RUS,2015q4,3400621706,3400.621706,High
1724754,KWT,2015q4,3333129856,3333.129856,High
1724769,KWT,2015q4,3193786592,3193.786592,High
1630702,FRA,2015q4,2874168874,2874.168874,High
1724760,SAU,2015q4,2785408765,2785.408765,High


## Analysis Results

The filtered dataset contains **127,272 records** based on the project's filtering rules.

The notebook uses the same imported project functions as `main.py`, so the analysis is based on the program's reusable modules rather than duplicating their implementation here.